go/Makefile

```
@@ -14,7 +14,7 @@ install-dev-tools-go1.22:
 	go install github.com/vakenbolt/go-test-report@v0.9.3
 	go install mvdan.cc/gofumpt@v0.6.0
 	go install github.com/segmentio/golines@v0.12.2
-	go install honnef.co/go/tools/cmd/staticcheck@v0.4.6
+	go install honnef.co/go/tools/cmd/staticcheck@v0.5.1
```

static check tools are by their nature always progressing and adding more tests and checks, I think we should simply use the latest version:

```suggestion
	go install honnef.co/go/tools/cmd/staticcheck@latest
```

---

Makes sense, done.